crates/database/src/auction_orders.rs

```diff
@@ -0,0 +1,68 @@
+use {
+    crate::{auction::AuctionId, OrderUid, PgTransaction},
+    sqlx::PgConnection,
+    std::ops::DerefMut,
+};
+
+/// Included order for a given auction.
+#[derive(Debug, Clone, PartialEq, sqlx::FromRow)]
+pub struct AuctionOrder {
+    pub auction_id: AuctionId,
+    pub order_uid: OrderUid,
+}
+
+pub async fn insert(
+    ex: &mut PgTransaction<'_>,
```

Comment:
IIRC a `PgTransaction` derefs into a `PgConnection`. Meaning if you want to have the most flexible function you should take a `PgConnection` here. Unless you want to force the caller to use a transaction but I don't see why this function should know whether it makes sense to call it in a tx.

If the idea was to not be wasteful with DB connections by using a tx here you could also issue a single query that stores all entries using a query builder.

Reply:
IMO, with PgTransaction, it is clear that the operation is executed inside a TX and the commit happens outside.

Reply:
PgConnection it is